actix-server/src/waker_queue.rs

```diff
@@ -0,0 +1,84 @@
+use std::ops::Deref;
+use std::sync::Arc;
+
+use concurrent_queue::{ConcurrentQueue, PopError};
+use mio::{Registry, Token as MioToken, Waker};
+
+use crate::worker::WorkerHandle;
+
+/// waker token for `mio::Poll` instance
+pub(crate) const WAKER_TOKEN: MioToken = MioToken(usize::MAX);
+
+/// `mio::Waker` with a queue for waking up the `Accept`'s `Poll` and contains the `WakerInterest`
+/// the `Poll` would want to look into.
+pub(crate) struct WakerQueue(Arc<(Waker, ConcurrentQueue<WakerInterest>)>);
+
+impl Clone for WakerQueue {
+    fn clone(&self) -> Self {
+        Self(self.0.clone())
+    }
+}
+
+impl Deref for WakerQueue {
+    type Target = (Waker, ConcurrentQueue<WakerInterest>);
+
+    fn deref(&self) -> &Self::Target {
+        self.0.deref()
+    }
+}
+
+impl WakerQueue {
+    /// construct a waker queue with given `Poll`'s `Registry` and capacity.
+    ///
+    /// A fixed `WAKER_TOKEN` is used to identify the wake interest and the `Poll` needs to match
+    /// event's token for it to properly handle `WakerInterest`.
+    pub(crate) fn with_capacity(registry: &Registry, cap: usize) -> std::io::Result<Self> {
+        let waker = Waker::new(registry, WAKER_TOKEN)?;
+        let queue = ConcurrentQueue::bounded(cap);
+
+        Ok(Self(Arc::new((waker, queue))))
+    }
+
+    /// push a new interest to the queue and wake up the accept poll afterwards.
+    pub(crate) fn wake(&self, interest: WakerInterest) {
+        let (waker, queue) = self.deref();
+
+        // ToDo: should we handle error here?
```

Comment:
```suggestion
        // FIXME: should we handle error here?
```

Reply:
We rarely use the queue so the chance of overflow it is low. 
It can be changed to a unbounded queue as it has no impact on server performance